bin/check-all-gpg-keys-for-expiry.rb

```diff
@@ -0,0 +1,348 @@
+         description: 'Location of GPG home directory'
+
+  option :gpg,
+         short: '-e GPG_EXECUTABLE_PATH',
+         long: '--execute GPG_EXECUTABLE_PATH',
+         default: '/usr/bin/gpg',
+         description: 'Location of binary executable file (gpg)'
+
+  option :use_secret_key,
+         short: '-s (yes|no)',
+         long: '--use-secret-key (yes|no)',
+         default: 'yes',
+         description: 'Check secret key instead of public key'
+
+  option :do_not_display_json_message,
+         short: '-d (yes|no)',
+         long: '--do-not-display-json-message (yes|no)',
+         default: 'no',
+         description: 'Display JSON message instead of sending it to Sensu'
+
+  def initialize
+    super
+  end
+
+  # Return the number of days since epoch from a given epoch
+  def self.days_since_epoch(e)
+    epoch = Date.new(1970, 1, 1)
+    d = Time.at(e).to_date
+    (d - epoch).to_i
+  end
+
+  # Return --list-key if 'use_secret_key' is set to nil or no otherwise
+  # return --list-secret-key
+  def key_type_option
+    if config[:use_secret_key].nil? || config[:use_secret_key] == 'no'
+      '--list-key'
+    else
+      '--list-secret-key'
+    end
+  end
+
+  # Add an option 'home_directory' as an argument to GPG command line if it is
+  # specified.
+  def home_dir_option
+    if config[:home_directory].nil?
+      [config[:gpg]]
+    else
+      [config[:gpg] + " --homedir #{config[:home_directory]}"]
+    end
+  end
+
+  # Return all GPG keys' details by executing the GPG command
+  def fetch_all_gpg_keys
+    args = ['--with-colons', '--fixed-list-mode', '--lock-never', key_type_option]
+    gpg_cmd = home_dir_option + args
+
+    _cmd_out, _cmd_err, _status = Open3.capture3 gpg_cmd.join ' '
+  end
+
+  # Set the key's level to the appropriate level by checking its number of days left
+  def update_key_level(key)
+    num_of_days_left = key.num_of_days_left
+    case num_of_days_left.to_s
+    when /^-\d+$/
+      key.level = 2
+    when /^\d+$/
+      return key.level = 0 if num_of_days_left > config[:warning]
+      return key.level = 2 if num_of_days_left <= config[:critical]
+      return key.level = 1
+    else
+      return key.level = 3
+    end
+  end
+
+  # Return the number of days left using key's expiry date and today date
+  def calculate_num_of_days_left(key, today_epoch)
+    CheckAllGpgKeysForExpiry.days_since_epoch(key.expiry.to_i) - CheckAllGpgKeysForExpiry.days_since_epoch(today_epoch)
+  end
+
+  # Populate key's num of days left and its level
+  def populate_key_num_of_days_left_and_level(key, today_epoch)
+    if !key.expiry.nil?
+      key.num_of_days_left = calculate_num_of_days_left(key, today_epoch)
+      update_key_level(key)
+    else
+      key.num_of_days_left = Integer::MAX
+      key.level = 0
+    end
+  end
+
+  # Return pub if 'use_secret_key' is set to nil or no otherwise
+  # return sec
+  def choose_key_type
+    if config[:use_secret_key].nil? || config[:use_secret_key] == 'no'
```

Comment:
why not bool?

Reply:
I was using old version of mixlib-cli which does not support boolean. I have updated `:use_secret_key` to use boolean.